tests/packages/test_main.py

```diff
@@ -183,6 +183,22 @@ def test_dependency_from_pep_508_with_git_url():
     assert "1.2" == dep.reference
 
 
+def test_dependency_from_pep_508_with_git_url_and_comment_and_extra():
+    name = (
+        "poetry @ git+https://github.com/python-poetry/poetry.git@b;ar;#egg=poetry"
+        ' ; extra == "foo;"'
+    )
+
+    dep = dependency_from_pep_508(name)
+
+    assert "poetry" == dep.name
+    assert dep.is_vcs()
+    assert "git" == dep.vcs
+    assert "https://github.com/python-poetry/poetry.git" == dep.source
+    assert "b;ar;" == dep.reference
```

Comment:
Is this a valid scenario?

Reply:
Yes, you can habe a branch with `;` in the name.
See https://github.com/python-poetry/core/blob/fbfcda73b2c5524c3eda29b1d9b70734a3a9dcf3/tests/packages/test_vcs_dependency.py#L55